nodejs/awsx/ec2/vpc.ts

```diff
@@ -145,32 +187,22 @@ export class Vpc extends pulumi.ComponentResource {
         createSubnets(vpc, name, "private", idArgs.privateSubnetIds);
         createSubnets(vpc, name, "isolated", idArgs.isolatedSubnetIds);
 
-        return vpc;
-    }
-}
+        if (idArgs.internetGatewayId) {
+            vpc.internetGateway = new x.ec2.InternetGateway(name, vpc, {
+                internetGateway: aws.ec2.InternetGateway.get(name, idArgs.internetGatewayId),
+            });
+        }
 
-function createInternetGateway(vpc: Vpc, name: string) {
-    if (vpc.publicSubnets.length === 0) {
-        return undefined;
-    }
+        if (idArgs.natGatewayIds) {
+            for (const natGatewayId of idArgs.natGatewayIds) {
+                vpc.natGateways.push(new x.ec2.NatGateway(name, vpc, {
```

Comment:
I think `name` needs a disambiguate suffix here - supplying two nat gateway IDs looks like it would create two NatGateway resources with the same URN.

Reply:
definitely!